net/dns/manager_test.go

```diff
@@ -591,12 +591,36 @@ func TestManager(t *testing.T) {
 				DefaultResolvers: mustRes("https://dns.nextdns.io/c3a884"),
 			},
 			os: OSConfig{
-				Nameservers: mustIPs("100.100.100.100"),
+				Nameservers: mustIPs("fd60:627a:a42b::53"),
 			},
 			rs: resolver.Config{
 				Routes: upstreams(".", "https://dns.nextdns.io/c3a884"),
 			},
 		},
+		{
+			name: "coder",
+			in: Config{
+				OnlyIPv6: true,
+				Routes: map[dnsname.FQDN][]*dnstype.Resolver{
+					"coder.": mustRes("fd60:627a:a42b::53"),
+				},
+				Hosts: hosts(
+					"agent.myws.me.coder.", "fd60:627a:a42c::53",
+				),
+			},
+			os: OSConfig{
+				Nameservers: mustIPs("fd60:627a:a42b::53"),
+			},
+			rs: resolver.Config{
+				Routes: upstreams(
+					".", "",
+					"coder.", "fd60:627a:a42b::53",
```

Comment:
I don't totally understand the purpose of routes on `resolver.Config`

Reply:
It's for allowing the DNS service to query external DNS servers. We don't need it for our purposes since the local DNS will be authoritative for all `**.coder.` names, and we'll use split-DNS so that the OS only asks us for these names in the first place.

For OSes which don't support split DNS, Tailscale becomes the default resolver, and then gets configured with upstream DNS servers for queries it can't handle locally.  We don't need to do this so we won't ever have upstream routes.